pkg/venafi/cloud/connector.go

```diff
@@ -757,6 +740,183 @@ func (c *Connector) SearchCertificate(zone string, cn string, sans *certificate.
+	err = c.validateIfCertIsVCPGeneratedByID(*(reqData.CertificateId))
+	if err != nil {
+		return nil, err
+	}
+	log.Println("Certificate is valid for provisioning (VCP generated)")
+
+	if reqData.KeystoreId == nil && (reqData.KeystoreName != nil && reqData.ProviderName != nil) {
+		log.Println("Keystore ID was not provided. Trying to find out by using keystore and provider names")
+		// TODO: Add logic to get keystore ID from Keystore's Name and its Provider Name once filters are ready
+		// here should feed up "keystoreId" var for next validation
+		if reqData.KeystoreId == nil {
+			return nil, fmt.Errorf("could not find Keystore ID for Keystore Name %s and Cloud Provider Name %s", *(req.KeystoreName), *(req.ProviderName))
+		}
+	}
+	keystoreIdString := *(reqData.KeystoreId)
+	log.Printf("Keystore ID for provisioning: %s", keystoreIdString)
+
+	if reqData.CertificateId == nil {
+		certId, err := c.getCertIDFromPickupID(*(reqData.PickupId), reqData.Timeout)
+		if err != nil {
+			return nil, err
+		}
+		reqData.CertificateId = certId
+	}
+	certificateIdString := *(reqData.CertificateId)
+	log.Printf("Certificate ID for provisioning: %s", certificateIdString)
+
+	// setting options for provisioning
+	log.Println("setting provisioning options")
+	provisioningOptions, err := setProvisioningOptions(options)
+	if err != nil {
+		return nil, err
+	}
+	log.Println("provisioning options successfully set")
+
+	graphqlClient := c.getGraphqlClient()
+
+	ctx := context.Background()
+
+	// Getting Keystore to find type
+	log.Printf("fetching keystore information for KeystoreID: %s", keystoreIdString)
+	data, err := cloudkeystores.GetCloudKeystoresByKeystoreId(ctx, graphqlClient, req.KeystoreId)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(data.CloudKeystores.Nodes) == 0 {
+		return nil, fmt.Errorf("could not find keystore with ID: %s", keystoreIdString)
+	}
+
+	if len(data.CloudKeystores.Nodes) > 1 {
+		return nil, fmt.Errorf("found more than one keystore for ID: %s", keystoreIdString)
+	}
+
+	// grabbing found keystore for later
+	cloudKeystore := data.CloudKeystores.Nodes[0]
+	log.Printf("successfully fetched keystore information for KeystoreID: %s", keystoreIdString)
+
+	wsClientId := uuid.New()
+
+	wsClientIdString := wsClientId.String()
+
+	wsConn, err := websocket.Subscribe(c.apiKey, c.baseURL, wsClientIdString)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Printf("Provisioning Certificate ID %s for Keystore %s", certificateIdString, keystoreIdString)
+	_, err = cloudkeystores.ProvisionCertificate(ctx, graphqlClient, certificateIdString, keystoreIdString, wsClientIdString, provisioningOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	ar, err := websocket.ReadResponse(wsConn)
+	if err != nil {
+		return nil, err
+	}
+
+	// parsing metadata from websocket response
+	log.Printf("Getting Cloud Metadata of Certificate ID %s and Keystore ID: %s", certificateIdString, keystoreIdString)
+	respMap := ar.Data.Result
```

Comment:
you are not doing anything with this variable, just pass `ar.Data.result` to `getCloudMetadataFromWebSocketResponse()` directly

Reply:
Resolved by latest commit